File: dotnet/Allmark/Utils/NewNode.cs
namespace Allmark;

using Allmark.Types;

public static partial class Utils
{
	// TODO: Move a lot of functionality into here
	// e.g. checking maybeContinue, checking if last node needs closing etc

	public static MarkdownNode NewNode(string type, bool block, int index, int line, int column, string markup, int indent, List<MarkdownNode>? children = null)
	{
		return new MarkdownNode
		{
			Type = type,
			Block = block,
			Index = index,
			Line = line,
			Column = column,
			Markup = markup,
			Delimiter = "",
			Content = "",
			Indent = indent,
			Subindent = 0,
			AcceptsContent = false,
			MaybeContinuing = false,
			BlankAfter = false,
			Children = children
		};
	}
}

File: zig/src/utils/newNode.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;

pub fn newNode(
    allocator: std.mem.Allocator,
    type_str: []const u8,
    block: bool,
    index: usize,
    line: i32,
    column: i32,
    markup: []const u8,
    indent: i32,
    children: ?[]*MarkdownNode,
) !*MarkdownNode {
    const node = try allocator.create(MarkdownNode);
    node.* = MarkdownNode{
        .type = try allocator.dupe(u8, type_str),
        .block = block,
        .index = index,
        .line = line,
        .column = column,
        .markup = try allocator.dupe(u8, markup),
        .delimiter = "",
        .content = "",
        .indent = indent,
        .subindent = 0,
        .acceptsContent = false,
        .maybeContinuing = false,
        .blankAfter = false,
        .info = null,
        .title = null,
        .children = children,
    };
    return node;
}
